File: String/String Compression.py
class Solution:
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        ptr, wptr = 0, 0
        while ptr < len(chars):
            ch, f = chars[ptr], 0
            while ptr < len(chars) and chars[ptr] == ch:
                ptr, f = ptr+1, f+1
            chars[wptr], wptr = ch, wptr+1
            if f > 1:
                for i in str(f):
                    chars[wptr], wptr = i, wptr+1
        return wptr

File: String/String Compression.cpp
class Solution {
	public:
    	int compress(vector<char>& chars) {
        	int ptr = 0, wptr = 0;
        	while(ptr < chars.size()){
            	char ch = chars[ptr];
            	int f = 0;
            	while(ptr < chars.size() && chars[ptr] == ch){
                	ptr++;
                	f++;
            	}
            	chars[wptr] = ch;
            	wptr++;
            	if(f > 1){
                	string s = to_string(f);
                	for(char c: s){
                    	chars[wptr] = c;
                    	wptr++;
                	}
            	}
        	}
        	return wptr;
    	}
	};